src/views/MainPage/components/UserContents.tsx

```diff
@@ -1,91 +1,121 @@
 
 import { APPLY_STATUS } from '../constants/applyStatus';
 import { DesignerResponse, ModelResponse } from '../hooks/type';
 
-import { OfferCard, ApplicationCard } from './Card';
+import { Card } from './Card';
 
-interface UserContentsProps {
-  data: DesignerResponse | ModelResponse;
+interface DesignerContentsProps {
+  data: DesignerResponse;
+  setPage: React.Dispatch<React.SetStateAction<number>>;
+}
+interface ModelContentsProps {
+  data: ModelResponse;
   setPage: React.Dispatch<React.SetStateAction<number>>;
 }
 
-const ReceivedOffer = (props: UserContentsProps) => {
-  const { data, setPage } = props;
-  const [ref, inView] = useInView();
+interface UserContentsProps {
+  title: string;
+  children: React.ReactNode;
+  ref: React.LegacyRef<HTMLDivElement>;
+  dataLength: number;
+  total: number;
+}
 
-  useEffect(() => {
-    inView && setPage((prev) => prev + 1);
-  }, [inView]);
+interface UserContentsProps {
+  title: string;
+  children: React.ReactNode;
+  dataLength: number;
+  total: number;
+}
 
-  return (
+const UserContents = React.forwardRef<HTMLDivElement, UserContentsProps>(
+  ({ title, children, dataLength, total }, ref) => (
     <S.UserContentsLayout>
-      <S.TitleSpan>도착한 제안서</S.TitleSpan>
-      {'status' in data && data.status === APPLY_STATUS.RECEIVED ? (
-        <S.ContentsBox>
-          {data.offers.map((offer, index) => (
-            <OfferCard
-              key={index}
-              offerId={offer.offerId}
-              name={offer.name}
-              shopName={offer.shopName}
-              imgUrl={offer.imgUrl}
-              isClicked={offer.isClicked}
-              conditions={offer.conditions}
-            />
-          ))}
-        </S.ContentsBox>
-      ) : (
-        <S.EmptyBox>
-          <S.HelperTextSpan>
-            {'status' in data && data.status === APPLY_STATUS.NOTHING ? (
-              '지금 바로 헤어모델에 지원해 보세요 :)'
-            ) : (
-              <S.LineHeightSpan>
-                조금만 기다리면 <br /> 디자이너의 모델 제안을 문자로 보내드릴게요 :&#41;
-              </S.LineHeightSpan>
-            )}
-          </S.HelperTextSpan>
-        </S.EmptyBox>
-      )}
-      {'offers' in data && data.offers.length < data.total && <div ref={ref}></div>}
+      <S.TitleSpan>{title}</S.TitleSpan>
+      <S.ContentsBox>{children}</S.ContentsBox>
+      {dataLength < total && <div ref={ref}></div>}
     </S.UserContentsLayout>
-  );
-};
+  ),
+);
+UserContents.displayName = 'UserContents';
 
-const ReceivedApplication = (props: UserContentsProps) => {
-  const { data, setPage } = props;
+const usePagination = (setPage: React.Dispatch<React.SetStateAction<number>>) => {
   const [ref, inView] = useInView();
 
   useEffect(() => {
-    inView && setPage((prev) => prev + 1);
-  }, [inView]);
+    if (inView) setPage((prev) => prev + 1);
+  }, [inView, setPage]);
+
+  return ref;
+};
+
+const ModelContents = ({ data, setPage }: ModelContentsProps) => {
+  const ref = usePagination(setPage);
+
```

Comment:
위의 타입을 바꿔주면 여기서 data가 undefined일 수 있다는 타입에러가 발생하겠죠? 

만약 ModelContents와 DesignerContents에서 렌더링 부에서만 data를 사용한다면 조건부 렌더링으로 구현해주면 되는데 리턴문 외에도 data를 많이 사용하고 있어서 빠른 리턴 코드 한줄 추가해서 타입에러들을 없앨 수 있습니다. 

```suggestion
if (!data) return;
```

어차피 넘겨받는 data는 일반 변수가 아닌 state이기 때문에 data가 비어있는 동안에는 ~Contents 내 코드를 실행하지 않았다가, data가 들어오면 그때 정상적인 호출 및 렌더링이 잘 진행되겠죠??

이런 방법도 한번 괜찮은지 고려해보면 좋을 것 같습니다 😎